lib/registry.js

```diff
@@ -16,7 +17,9 @@ export function define(id, spec) {
   if (get(id)) {
     console.warn(`Overwriting spec ${getName(id)}`)
   }
-  spec.name = getName(id)
+  if (!spec.name && typeof spec !== "symbol") {
```

Comment:
My PR includes predicates (easy) and enforced Symbol IDs in the registry. It would be straightforward to convert the PR to enforce String IDs instead.